### ISSUE TITLE
GRANT 'role' TO 'user' statement not replicating across nodes

### ISSUE THREAD

**Issue by janlindstrom** [mariadb-corporation/mysql-wsrep#421]
Originally reported by @milkmanbilly

```
OS: 
Ubuntu 22.04.4 LTS

Packages:
galera-4_ 26.4.20+1jammy
mysql-wsrep_8.4.2-26.20+1ubuntu22.04
```
Granting roles to users (https://dev.mysql.com/doc/refman/8.4/en/grant.html#grant-roles) are not being replicated across galera nodes.

Steps to reproduce:

**Node A**
```
mysql> CREATE ROLE backup_role;
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT user AS role_name FROM mysql.user WHERE host = '%' AND NOT LENGTH(authentication_string);
+-------------------------+
| role_name               |
+-------------------------+
| backup_role             |
+-------------------------+
1 rows in set (0.00 sec)

mysql> GRANT BACKUP_ADMIN,FLUSH_PRIVILEGES ON *.* TO backup_role;
Query OK, 0 rows affected (0.00 sec)

mysql> show grants for backup_role;
+-----------------------------------------------------------------+
| Grants for backup_role@%                                        |
+-----------------------------------------------------------------+
| GRANT USAGE ON *.* TO `backup_role`@`%`                         |
| GRANT BACKUP_ADMIN,FLUSH_PRIVILEGES ON *.* TO `backup_role`@`%` |
+-----------------------------------------------------------------+
2 rows in set (0.00 sec)

mysql> GRANT backup_role TO backup;
Query OK, 0 rows affected (0.00 sec)

mysql> show grants for backup;
+------------------------------------------------------------------------------------------------+
| Grants for backup@%                                                                            |
+------------------------------------------------------------------------------------------------+
| GRANT SELECT, RELOAD, PROCESS, LOCK TABLES, EXECUTE, REPLICATION CLIENT ON *.* TO `backup`@`%` |
| GRANT `backup_role`@`%` TO `backup`@`%`                                                        |
+------------------------------------------------------------------------------------------------+
2 rows in set (0.00 sec)
```

**Node B** (Every other GRANT <priv> or CREATE <user/role> statements are replicated without any issue)

```
mysql> SELECT user AS role_name FROM mysql.user WHERE host = '%' AND NOT LENGTH(authentication_string);
+-------------------------+
| role_name               |
+-------------------------+
| backup_role             |
+-------------------------+
1 rows in set (0.00 sec)

mysql> show grants for backup_role;
+-----------------------------------------------------------------+
| Grants for backup_role@%                                        |
+-----------------------------------------------------------------+
| GRANT USAGE ON *.* TO `backup_role`@`%`                         |
| GRANT BACKUP_ADMIN,FLUSH_PRIVILEGES ON *.* TO `backup_role`@`%` |
+-----------------------------------------------------------------+
2 rows in set (0.00 sec)

mysql> show grants for backup;
+------------------------------------------------------------------------------------------------+
| Grants for backup@%                                                                            |
+------------------------------------------------------------------------------------------------+
| GRANT SELECT, RELOAD, PROCESS, LOCK TABLES, EXECUTE, REPLICATION CLIENT ON *.* TO `backup`@`%` |
+------------------------------------------------------------------------------------------------+
1 rows in set (0.00 sec)
```

**Comment by temeo** [comment]
@milkmanbilly Thanks for the report!

The original report was accidentally transferred to wrong repository, and github does not allow to transfer it back to public repository. Recreated the issue here.

**Comment by milkmanbilly** [comment]
I have tested it on version 8.0.37 and it is also failing. Maybe the fix can be implemented in version 8.0 as well.

**Comment by 4durban** [comment]
Regards!

Have you been able to solve the bug? I am also interested in this but I can not find any PR or where I can check if it has been solved or not. I guessed it has been solved, since it is marked as completed, but would you mind providing further indications in order to know when or if this will get solved?

Sorry for bothering and thank you for your time!

**Comment by temeo** [comment]
The fix for this issue has been released in 8.0.41-26.22.

**Comment by milkmanbilly** [comment]
Thank you for everything
